.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,203 @@
+name: Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  SKIP_ENV_VALIDATION: "true"
+
+jobs:
+  changed-packages:
+    name: Determine which apps changed
+    uses: ./.github/workflows/changed-packages.yml
+
+  lint-admin-ui:
+    name: Lint admin ui
+    runs-on: ubuntu-latest
+    needs: [changed-packages]
+    # TODO disabled for testing
+    if: needs.changed-packages.outputs.admin_changed == 'true'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+      - name: Lint admin ui
+        run: |
+          pnpm lint --filter=admin-ui...
+          pnpm lint:css --filter=admin-ui...
+          # TODO run tsc also when it's not broken
+
+  build-admin-ui:
```

Comment:
For Docker build pipelines, you could use the ready-made Docker actions and push the UI containers to [ghcr.io](ghcr.io). [See how I've done it in the backend](https://github.com/City-of-Helsinki/tilavarauspalvelu-core/blob/main/.github/workflows/build.yml). This allows you to store the layer cache in the registry between builds.

Note that the registry needs to be created manually on the first time for the project by pushing there, then github actions can push automatically.